src/py21cmfast/inputs.py

```diff
@@ -0,0 +1,614 @@
+        unity for a flat universe.
+    Y_He : float
+        Helium fraction.
+    wl : float
+        Dark energy equation of state parameter (wl = -1 for vacuum )
+    SHETH_b : float
+        Sheth-Tormen parameter for ellipsoidal collapse (for HMF).
+
+        .. note:: The best fit b and c ST params for these 3D realisations have a redshift,
+                  and a DELTA_R_FACTOR (see ANAL_PARAMS.H) dependence, as shown
+                  in Mesinger+. For converged mass functions at z~5-10, set `DELTA_R_FACTOR=1.1`
+                  and `SHETH_b=0.15` and `SHETH_c~0.05`.
+
+                  For most purposes, a larger step size is quite sufficient and provides an
+                  excellent match to N-body and smoother mass functions, though the b and c
+                  parameters should be changed to make up for some "stepping-over" massive
+                  collapsed
+                  halos (see Mesinger, Perna, Haiman (2005) and Mesinger et al.,
+                  in preparation).
+
+                  For example, at z~7-10, one can set `DELTA_R_FACTOR=1.3` and `SHETH_b=0.15`
+                   and `SHETH_c=0.25`, to increase the speed of the halo finder.
+    SHETH_c : float
+        Sheth-Tormen parameter for ellipsoidal collapse (for HMF). See notes for `SHETH_b`.
+    Zreion_HeII : float
+        Redshift of helium reionization, currently only used for tau_e
+    FILTER : int, {0, 1}
+        Filter to use for smoothing.
+        0. tophat
+        1. gaussian
+    external_table_path : str
+        The system path to find external tables for calculation speedups. DO NOT MODIFY.
+    """
+
+    def __init__(self, wrapped, ffi):
+        super().__init__(wrapped, ffi)
+
+        EXTERNALTABLES = ffi.new(
+            "char[]", path.join(path.expanduser("~"), ".21cmfast").encode()
+        )
+        self.external_table_path = EXTERNALTABLES
+
+
+global_params = GlobalParams(lib.global_params, ffi)
+
+
+class CosmoParams(StructWithDefaults):
+    """
+    Cosmological parameters (with defaults) which translates to a C struct.
+
+    To see default values for each parameter, use ``CosmoParams._defaults_``.
+    All parameters passed in the constructor are also saved as instance attributes which should
+    be considered read-only. This is true of all input-parameter classes.
+
+    Parameters
+    ----------
+    SIGMA_8 : float, optional
+        RMS mass variance (power spectrum normalisation).
+    hlittle : float, optional
+        The hubble parameter, H_0/100.
+    OMm : float, optional
+        Omega matter.
+    OMb : float, optional
+        Omega baryon, the baryon component.
+    POWER_INDEX : float, optional
+        Spectral index of the power spectrum.
+    """
+
+    _ffi = ffi
+
+    _defaults_ = {
+        "SIGMA_8": 0.82,
+        "hlittle": Planck15.h,
+        "OMm": Planck15.Om0,
+        "OMb": Planck15.Ob0,
+        "POWER_INDEX": 0.97,
```

Comment:
Why don't we set SIGMA_8 and and POWER_INDEX to the Planck 2015 values?

Reply:
We can do that, but it should be in another PR. I'll make an issue for it now.